src/main/java/be/syntra/devshop/DevshopBack/services/SearchServiceImpl.java

```diff
@@ -0,0 +1,85 @@
+package be.syntra.devshop.DevshopBack.services;
+
+import be.syntra.devshop.DevshopBack.entities.Product;
+import be.syntra.devshop.DevshopBack.models.SearchModel;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Slf4j
+@Service
+public class SearchServiceImpl implements SearchService {
+    private SearchModel searchModel = null;
+    private ProductService productService;
+
+    @Autowired
+    public SearchServiceImpl(ProductService productService) {
+        this.productService = productService;
+    }
+
+    @Override
+    public void setSearchModel(SearchModel searchModel) {
+        this.searchModel = searchModel;
+    }
+
+    @Override
+    public List<Product> applySearchModel(SearchModel searchModel) {
+        final Comparator<Product> productNameComparator = (searchModel.isSortAscendingName())
+                ? Comparator.comparing(Product::getName)
+                : Comparator.comparing(Product::getName).reversed();
+        final Comparator<Product> productPriceComparator = (searchModel.isSortAscendingPrice())
+                ? Comparator.comparing(Product::getPrice)
+                : Comparator.comparing(Product::getPrice).reversed();
+        List<Product> productList;
+
+        if (searchModel.isArchivedView()) {
+            productList = productService.findAllByArchivedTrue();
+        } else {
+            productList = productService.findAllByArchivedFalse();
+        }
+
+        if (null != searchModel.getSearchRequest()) {
```

Comment:
if (StringUtils.hasText(searchModel.getSearchRequest())) {

Reply:
didn't know that, very handy, fixed